certbot-nginx/certbot_nginx/nginxparser.py

```diff
@@ -2,85 +2,53 @@
 # Forked from https://github.com/fatiherikli/nginxparser (MIT Licensed)
 import copy
 import logging
-import string
 
 from pyparsing import (
-    Literal, White, Word, alphanums, CharsNotIn, Combine, Forward, Group,
-    Optional, OneOrMore, Regex, ZeroOrMore)
+    Literal, White, Forward, Group, Optional, OneOrMore, Regex, ZeroOrMore, Combine)
 from pyparsing import stringEnd
 from pyparsing import restOfLine
 
 logger = logging.getLogger(__name__)
 
 class RawNginxParser(object):
     # pylint: disable=expression-not-assigned
+    # pylint: disable=pointless-statement
     """A class that parses nginx configuration with pyparsing."""
 
     # constants
     space = Optional(White())
-    nonspace = Regex(r"\S+")
+
     left_bracket = Literal("{").suppress()
     right_bracket = space.leaveWhitespace() + Literal("}").suppress()
     semicolon = Literal(";").suppress()
-    key = Word(alphanums + "_/+-.")
-    dollar_var = Combine(Literal('$') + Regex(r"[^\{\};,\s]+"))
-    condition = Regex(r"\(.+\)")
-    # Matches anything that is not a special character, and ${SHELL_VARS}, AND
-    # any chars in single or double quotes
-    # All of these COULD be upgraded to something like
-    # https://stackoverflow.com/a/16130746
     dquoted = Regex(r'(\".*\")')
     squoted = Regex(r"(\'.*\')")
```

Comment:
I know these two definitely are unchanged from before, but could this be inappropriately greedy where the quoted string includes other things in the middle, like

`"hello"; # now you don't notice this is a comment "heh heh"`

or something? (`dquoted` would match that entire line, if it's allowed to.) I know the parser has ways of handling this through prioritization of the things that are built up out of these elements, but was the context in which `dquoted` and `squoted` get used correct in this regard before, and is it still correct now?

A possibly safer alternative would be

`dquoted = Regex(r'(\"[^"]*\")')`

`squoted = Regex(r"(\'[^']*\')")`